swap/config.go

```diff
@@ -34,6 +34,6 @@ const (
 	// The smart-contract allows for setting this variable differently per beneficiary
 	defaultHarddepositTimeoutDuration = 24 * time.Hour
 
-	// While Swap is unstable, it's only allowed to be run under a specific network ID
-	AllowedNetworkID = 5
+	// While Swap is unstable, it is not allowed to run on mainnet
+	DisabledNetworkID = 1
```

Comment:
you are making a change here which i don't believe the PR nor the original issue specify.

please provide clarification

Reply:
Correct. Do you think I should leave it out?

This change was made because:
- There were to variables `DefaultNetworkID`(`network/protocol.go`)and AllowedNetworkId. To make things even stranger, is that the `DefaultNetworkID`, was not the same as the `AllowedNetworkID`. 
- I think the purpose of this variable here, is to make sure Swarm is not ran on main net, which is much better to express like this. 
- Furthermore, I sometimes want to test Swarm on a different testnet than 5, and I had to change this all the time manually.

Reply:
thanks for the clarification. a few things:
- the main idea behind the original `AllowedNetworkID` var was to only allow SWAP on a specific network since it was—and still is—so unstable. this is why the `DefaultNetworkID` var is different, so as to disable SWAP by default, and enable it only when the user willingly indicates so.
- i agree with you that this is more expressive regarding its intent.
- i would verify this at least with @holisticode to make sure we want SWAP enabled everywhere except on the default network ID, or the inverse.
- i don't think we should have a separate issue for this. but: if we go ahead with this change, we should document it in the original/relevant issue.

Reply:
*"and enable it only when the user willingly indicates so"* —this is the case now as well, as the user must have set the `swapEnabled` flag, put a `swap-backend-URL` and he must have ethers on the testnet which he indicates to use. All this cannot really be done by accident, and even if it was an accident, it is not done on main net (due to `disabledNetworkID`), so there is no damage if an accident happens.

Willing to document this change in the description of the PR, awaiting feedback from @holisticode :)

Reply:
good points @Eknir. let's wait for additional feedback then :+1:

Reply:
Feedback from @holisticode  (to keep things here and because I can't give replies to general PR comments)
>As for the network IDs: The DefaultNetworkID indicates something different and independent from swap. It should not be considered when deciding where we enable swap or not.

>Now the tricky thing is that currently, if we deploy a new node, (as far as I know! To be confimed!!!) it would pick the DefaultNetworkID for deployment. We don't have NetworkID == 1 as mainnet yet! So all current nodes would connect to that network - in other words our network id 4 is our current mainnet!

>Thus, while the intention is correct, and the change would be correct once we have indeed a mainnet, I think it is not good to change this now. I would prefer to have to specifically enable the swap enabled test net - unless my assumptions are incorrect.

Reply:
Reverting the changes, as they were guided by my misunderstanding. I will explain what happened:

I thought the `NetworkID` variable was about the Ethereum network ID, where we do all the interactions with the chequebook. Obviously this is not the case, but I thought it was as I use myself the Ethereum Network ID 4 myself, and I could only make it work by setting the allowedNetworkID to 4. Now, I know better and I know that this is about the BZZ network ID, and I can set it myself via flag or config.

I updated a comment to prevent this confusion to happen again in the future. 

How we are going to handle this in the future, will be an important decision. But I think that indeed right now is not the time to think about that--let's first make SWAP stable!